main.go

```diff
@@ -1,51 +1,166 @@
 package main
 
 import (
+	"bufio"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
+	"io"
 	"os"
 
 	"github.com/yazgazan/jaydiff/diff"
 )
 
 const (
-	statusUsage          = 2
-	statusReadError      = 3
-	statusUnmarshalError = 4
-	statusDiffError      = 5
-	statusDiffMismatch   = 6
+	statusUsage           = 2
+	statusReadError       = 3
+	statusUnexpectedError = 4
+	statusDiffError       = 5
+	statusDiffMismatch    = 6
 )
 
 var (
 	// Version is replaced by the tag when creating a new release
 	Version = "dev"
 )
+type nextJson func() (interface{}, int, error)
 
 func main() {
-	var err error
+	var (
+		lhs_file,rhs_file *os.File
```

Comment:
I would like to avoid *snake_case* variables in this project, to keep everything consistent.